core/update.php

```diff
@@ -228,7 +228,7 @@ function update_results_page() {
   }
 
   if (settings_get('update_free_access')) {
-    $output .= "<p><strong>Reminder: Don't forget to set the <code>\$settings[&#39;update_free_access&#39;]</code> value in your <code>settings.php</code> file back to <code>FALSE</code>.</strong></p>";
+    backdrop_set_message("Reminder: Don't forget to set the <code>\$settings['update_free_access']</code> value in your <code>settings.php</code> file back to <code>FALSE</code>.", 'warning');
```

Comment:
Rather than escaping the variable, shouldn't this instead be:

```suggestion
    backdrop_set_message("Reminder: Don't forget to set the <code>" . $settings['update_free_access'] . "</code> value in your <code>settings.php</code> file back to <code>FALSE</code>.", 'warning');
```

Reply:
@BWPanda your code change would actually try to concatenate the value of the `$settings['update_free_access']` variable, which would throw "undefined variable" errors etc.

What we are trying to do here is to output the actual string `$settings['update_free_access']`. Since this string already contains single quotes, we cannot use single quotes in `backdrop_set_message()`, but using double quotes would try to interpret the string as code (a variable). Hence the escaping of the `$`.

I hope that this makes sense now.

The alternative would be something like this:

```suggestion
    $free_access_warning = format_string('Reminder: Don't forget to set the <code>@free_access</code> value in your <code>settings.php</code> file back to <code>FALSE</code>.', array('@free_access' => "$settings['update_free_access']"));
    backdrop_set_message($free_access_warning, 'warning');
```

...but too much code IYAM.

Reply:
Ah, ok, I get it now. Sorry, my bad.